docs/extensions.rst

```diff
@@ -77,3 +77,37 @@ and treat it as if the client had made a request.
 
 The ASGI server should set the pseudo ``:authority`` header value to
 be the same value as the request that triggered the push promise.
+
+Sendfile
+------------------
+
+Sendfile allows you to send the contents of a file descriptor to the
+HTTP client with zero copy. ASGI servers that implement this extension will
+provide ``http.response.sendfile`` in the extensions part of the scope::
+
+    "scope": {
+        ...
+        "extensions": {
+            "http.response.sendfile": {},
+        },
+    }
+
+An ASGI framework can call ``sendfile`` by sending a message with
+the following keys. This message can be sent at any time after the
+*Response Start* message but before the final *Response Body* message.
+
+Keys:
+
+* ``type`` (*Unicode string*): ``"http.response.sendfile"``
+
+* ``file`` (*file descriptor object*): An open file descriptor object:
+  like file or socket.
```

Comment:
I expected this to take a filename, not a file descriptor. What's the reasoning here?

Reply:
I *think* they're trying to say "file-like object", maybe to allow for returning `BytesIO`? However, if this is required for regular files too, it needs to be made clear what should be closing the file and when.

Reply:
> I expected this to take a filename, not a file descriptor. What's the reasoning here?

Using a file descriptor allows to bridge two TCP Sockets, not necessarily files.

Reply:
> I *think* they're trying to say "file-like object", maybe to allow for returning `BytesIO`? However, if this is required for regular files too, it needs to be made clear what should be closing the file and when.

The purpose of this extension is to allow zero-copy data transfer using the sendfile feature. BytesIO should uses a normal method to transfer.

Reply:
Yup, now I've read the appropriate asyncio source code, I see. Let's make the phrasing a bit clearer, then - maybe "a Python file object with an underlying OS file descriptor (e.g. not BytesIO)"

Reply:
Although I don't think the "file descriptor" is not clear enough, I will change it.